ospfd/ospfd.c

```diff
@@ -380,18 +414,50 @@ struct ospf *ospf_get_instance(u_short instance)
 
 	ospf = ospf_lookup_instance(instance);
 	if (ospf == NULL) {
-		ospf = ospf_new(instance);
+		ospf = ospf_new(instance, NULL /* VRF_DEFAULT*/);
 		ospf_add(ospf);
 
-		if (ospf->router_id_static.s_addr == 0)
+		if (ospf->router_id_static.s_addr == 0) {
+			if (vrf_lookup_by_id(ospf->vrf_id))
+				ospf_router_id_update(ospf);
+			else {
+				if (IS_DEBUG_OSPF_EVENT)
+					zlog_debug("%s: ospf VRF (id %d) is not active yet, skip router id update"
+						    , __PRETTY_FUNCTION__,
+						    ospf->vrf_id);
+			}
 			ospf_router_id_update(ospf);
+		}
 
 		ospf_opaque_type11_lsa_init(ospf);
 	}
 
 	return ospf;
 }
 
+struct ospf *ospf_lookup_by_vrf_id(vrf_id_t vrf_id)
+{
+	struct vrf *vrf = NULL;
+
+	vrf = vrf_lookup_by_id(vrf_id);
+	if (!vrf)
+		return NULL;
+	return (vrf->info) ? (struct ospf *)vrf->info : NULL;
+
+}
+
+struct ospf *ospf_lookup_by_name(const char *name)
```

Comment:
please make this function static and add a comment that it should only be used when processing incoming data from zebra, because in other situations it's wrong to lookup the ospf instance by name only without using the instance number.

Also please rename to `ospf_lookup_by_vrf_name` (`name` is ambiguous/misleading)

Reply:
I have moved to static and added comments as suggested.